### PR TITLE
[vectorized](function) support array_filter function

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

File: docs/en/docs/sql-manual/sql-functions/array-functions/array_filter.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+## array_filter
+
+<version since="1.2.2">
```
Comment (HappenLee) [inline docs/en/docs/sql-manual/sql-functions/array-functions/array_filter.md:29]: not 1.2.2 version

File: docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_filter.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+<version since="1.2.2">
+
+array_filter(lambda,array)
+
+</version>
+
+### description
+
+Use a lambda expression as the input parameter to calculate and filter the internal data of other input ARRAY parameters.
```
Comment (HappenLee) [inline docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_filter.md:37]: use chinese please

File: docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_filter.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+Filter out the values of 0 and NULL in the result, and only keep the values of 1.
+
+```
+array_filter(x->x>0, array1);
+array_filter(x->(x+2)=10, array1);
+array_filter(x->(abs(x)-2)>0, array1);
```
Comment (HappenLee) [inline docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_filter.md:43]: what will occur if the first lambda function do not return `boolean`? 
1. return error ?
2. do cast to boolean?

File: be/src/vec/exprs/lambda_function/varray_filter_function.cpp [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    static LambdaFunctionPtr create() { return std::make_shared<ArrayFilterFunction>(); }
+
+    std::string get_name() const override { return name; }
+
+    doris::Status execute(VExprContext* context, doris::vectorized::Block* block,
+                          int* result_column_id, DataTypePtr result_type,
+                          const std::vector<VExpr*>& children) override {
+        ///* array_filter(array, array<boolean>) *///
+
+        //1. child[0:end]->execute(src_block)
```
Comment (HappenLee) [inline be/src/vec/exprs/lambda_function/varray_filter_function.cpp:49]: the first step work also need to do in `array_map`, so it code should in  `LambdaFunction`

File: docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_filter.md (hunk 3 above)
Comment (zhangstar333) [reply docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_filter.md:43]: have a cast to bool

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run feut p0

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.86 seconds
 stream load tsv:          470 seconds loaded 74807831229 Bytes, about 151 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          76 seconds loaded 1101869774 Bytes, about 13 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230317151505_clickbench_pr_116168.html

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**